### PR TITLE
Add setting "output_format_pretty_max_value_width".

### LINKED ISSUE #11140: Cannot quit from clickhouse-client with Ctrl+C.
```
milovidov-desktop :) SELECT arrayJoin(arrayMap(y -> [arrayMap(x -> (cos(x / 90. * pi()) * y, sin(x / 90. * pi()) * y), range(180))], arraySlice(range(35000), 2, 35000)))

SELECT arrayJoin(arrayMap(y -> [arrayMap(x -> (cos((x / 90.) * pi()) * y, sin((x / 90.) * pi()) * y), range(180))], arraySlice(range(35000), 2, 35000)))

^C^C^C^C^C^C^C^C^C^C^C^C^C^C^C^C^C^C^C^C
```

### PR DESCRIPTION
**Pull request by alexey-milovidov**
Changelog category (leave one):
- Improvement

Changelog entry (a user-readable short description of the changes that goes to CHANGELOG.md):
Add setting "output_format_pretty_max_value_width". If value is longer, it will be cut to avoid output of too large values in terminal. This closes #11140 

Detailed desciption:
Enhancement of #3003

### CODE REVIEW

File: src/Common/UTF8Helpers.cpp [hunk 1]
```diff
@@ -87,44 +89,118 @@ static int wcwidth(wchar_t wc)
...
+#endif
+
+        while (i < size && isPrintableASCII(data[i]))
+        {
+            ++width;
+            ++i;
+        }
+
+        if (mode == BytesBeforLimit && width > limit)
+            return i - (width - limit);
```
Comment (KochetovNicolai) [inline src/Common/UTF8Helpers.cpp:147]: Too difficult to understand how do we calculate it.
`i`  is `size` now. Why don't we use size?
As I understood, we need to return the number of bytes to print `limit` printable ascii symbols.
`(with - limit)` is the number of printable symbols after limit. So, `size - (with - limit)` >= the number of bytes we need to return.
Are we ok with not exact result? If so, lets' add comment.
Also, why not to check that we have enough printable symbols in a loop?

Comment (KochetovNicolai) [reply src/Common/UTF8Helpers.cpp:147]: Example is: `x*x*x*x******************xxx`, where `x` is printable and `*` is not, limit is '4'.  `size - (with - limit)` is `size - (7-4) = size - 3`, but 8 bytes enough.

Comment (alexey-milovidov) [reply src/Common/UTF8Helpers.cpp:147]: > i is size now.

No, i is the first position after a range of consecutive printable ASCII characters.

> As I understood, we need to return the number of bytes to print limit printable ascii symbols.

No, we need to return the number of bytes to print unicode glyphs that span no more than `limit` of visible width.

Comment (alexey-milovidov) [reply src/Common/UTF8Helpers.cpp:147]: > (with - limit)

`width` not `with`.

Comment (alexey-milovidov) [reply src/Common/UTF8Helpers.cpp:147]: > So, `size - (with - limit)` >= the number of bytes we need to return.

But we use `i - (width - limit)`

`width - limit` is the number of "extra" printable ASCII characters and `i` is the number of bytes including this "extra".

Comment (alexey-milovidov) [reply src/Common/UTF8Helpers.cpp:147]: > Are we ok with not exact result?

No, we always return exact result.

For example, we have string "你好", it has two code points and visible width of 4 and byte size of 6.

Suppose we have limit = 3.
Then we have to return 3 as maximum number of bytes and the truncated string will be "你": one code point, visible width 2, byte size 3.

Comment (alexey-milovidov) [reply src/Common/UTF8Helpers.cpp:149]: I have added these comments.

Comment (KochetovNicolai) [reply src/Common/UTF8Helpers.cpp:149]: Ok, I've got it.
Just have read this code wrongly at first time.

File: tests/queries/0_stateless/00305_http_and_readonly.reference [hunk 2]
```diff
@@ -1,19 +1,19 @@
-name               value   changed
+ name               value   changed
```
Comment (KochetovNicolai) [inline tests/queries/0_stateless/00305_http_and_readonly.reference:1]: This change looks suspicious, but probably it's ok for `PrettySpace`.
Hopefully it won't brake somebody's parser.

**Review by KochetovNicolai — APPROVED** [review]

File: tests/queries/0_stateless/00305_http_and_readonly.reference (hunk 2 above)
Comment (alexey-milovidov) [reply tests/queries/0_stateless/00305_http_and_readonly.reference:1]: Yes, this whitespace is to make code simpler :)

**Comment by alexey-milovidov** [comment]
`test_ttl_move/test.py::test_concurrent_alter_with_ttl_move[concurrently_altering_ttl_mt-MergeTree()]`

- flaky test, fixed in #11355

**Pull request merged**